common/lib/capa/capa/responsetypes.py

```diff
@@ -831,16 +841,110 @@ def assign_choice_names(self):
             if not choice.get('id'):
                 choice.set("id", chr(ord("A") + index))
 
+    def grade_via_edc(self, all_choices, student_answer, student_non_answers):
+        """
+        Calculates partial credit on the EDC scheme.
+        For each correctly selected or correctly blank choice, score 1 point.
+        Divide by total number of choices.
+        Returns a CorrectMap.
```

Comment:
I think "edc" is fine in variable names, but I would agree that the long
form should be used in the docstring, and potentially the function name as
well. Thanks for that nit, Piotr: it's a good point, most won't be familiar
with the acronym.

On Wed, Jun 24, 2015 at 5:47 PM, Piotr Mitros <notifications@github.com>
wrote:

> In common/lib/capa/capa/responsetypes.py
> <https://github.com/edx/edx-platform/pull/8574#discussion_r33200816>:
>
> > @@ -831,16 +841,110 @@ def assign_choice_names(self):
> >              if not choice.get('id'):
> >                  choice.set("id", chr(ord("A") + index))
> >
> > +    def grade_via_edc(self, all_choices, student_answer, student_non_answers):
> > +        """
> > +        Calculates partial credit on the EDC scheme.
> > +        For each correctly selected or correctly blank choice, score 1 point.
> > +        Divide by total number of choices.
> > +        Returns a CorrectMap.
>
> Nit: Either in the function name or in the function comment, use the
> non-acronym version of edc. I like 'edc' in the XML, since it is concise,
> but in the code, it should be clear.
>
> I usually put things like this in function names. I code with 'edc' and
> then do a search-and-replace before committing, but that's a question of
> personal style.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/8574/files#r33200816>.
>